waiter/src/waiter/handler.clj

```diff
@@ -377,12 +401,41 @@
                                            scheduler-interactions-thread-pool))]
             (if error
               (utils/exception->response error request)
-              (let [delete-result result
-                    response-status (case (:result delete-result)
+              (let [delete-result (:result result)
+                    response-status (case delete-result
                                       :deleted http-200-ok
                                       :no-such-service-exists http-404-not-found
                                       http-400-bad-request)
-                    response-body-map (merge {:success (= http-200-ok response-status), :service-id service-id} delete-result)]
+                    router-id->response-chan (when (and (= http-200-ok response-status)
+                                                        (pos? timeout))
+                                               (assoc
+                                                 (make-inter-router-requests-fn (str "apps/" service-id "/await-deletion")
+                                                                                :method :get
+                                                                                :config {:query-string (str "timeout=" timeout)})
+                                                 router-id (async/go
+                                                             {:body (async/go
+                                                                      (json/write-str
+                                                                        {:exists? (async/<! (await-service-deletion-locally
+                                                                                              fallback-state-atom service-id timeout 100))}))})))
+                    router-id->exists? (loop [result {}
+                                              [[router-id response-chan] & remaining] (seq router-id->response-chan)]
+                                         (if (and router-id response-chan)
+                                           (recur
+                                             (assoc result router-id (-> response-chan
```

Comment:
```suggestion
                                             (assoc result router-id (some-> response-chan
```
This protects us against the scenario when the request fails and there is no body available.